### PR TITLE
disk: handle when passed disktype is empty

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/schema.py [hunk 1]
```diff
@@ -473,7 +473,18 @@ class DiskOptionSettings(FeatureSettings):
             ],
         ),
         metadata=field_metadata(
-            decoder=partial(search_space.decode_set_space_by_type, base_type=DiskType)
+            decoder=lambda input: (
+                search_space.decode_set_space_by_type(data=input, base_type=DiskType)
```
Comment (squirrelsc) [inline lisa/schema.py]: Write a method, it would be clearer than lambda.

Comment (LiliDeng) [reply lisa/schema.py]: updated

File: lisa/search_space.py [hunk 2]
```diff
@@ -710,3 +710,15 @@ def create_set_space(
     else:
         set_space = None
     return set_space
+
+
+def decode_empty_count_space(
```
Comment (squirrelsc) [inline lisa/search_space.py]: It should be called like `decode_nullable_set_space`. It's not a count space or always empty one.

File: lisa/search_space.py [hunk 3]
```diff
@@ -710,3 +710,15 @@ def create_set_space(
...
+    data: Any, base_type: Any, allowed_values: Any, is_allow_set: bool = False
```
Comment (squirrelsc) [inline lisa/search_space.py]: `allowed_values` should be `default_values`

File: lisa/search_space.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/search_space.py]: updated

File: lisa/search_space.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/search_space.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**